enterprise/server/raft/store/store.go

```diff
@@ -1314,6 +1320,38 @@ func (s *Store) SplitCluster(ctx context.Context, req *rfpb.SplitClusterRequest)
 	return splitRsp, nil
 }
 
+func (s *Store) getLastAppliedIndex(header *rfpb.Header) (uint64, error) {
+	r, _, err := s.validatedRange(header)
+	if err != nil {
+		return 0, err
+	}
+	return r.LastAppliedIndex()
+}
+
+func (s *Store) waitForReplicaToCatchUp(ctx context.Context, clusterID uint64, desiredLastAppliedIndex uint64) error {
+	start := time.Now()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			break
+		}
+		if rd := s.lookupRange(clusterID); rd != nil {
+			if r, err := s.GetReplica(rd.GetRangeId()); err == nil {
+				if lastApplied, err := r.LastAppliedIndex(); err == nil {
```

Comment:
Should you return an error if either GetReplica or LastAppliedIndex fails? 
They should only error out in catastrophic cases, right?